cli/admin/src/main/java/io/pravega/cli/admin/utils/SerializerUtils.java

```diff
@@ -0,0 +1,117 @@
+                    .put("length", SegmentMetadata::getLength)
+                    .put("chunkCount", SegmentMetadata::getChunkCount)
+                    .put("startOffset", SegmentMetadata::getStartOffset)
+                    .put("status", SegmentMetadata::getStatus)
+                    .put("maxRollingLength", SegmentMetadata::getMaxRollinglength)
+                    .put("firstChunk", SegmentMetadata::getFirstChunk)
+                    .put("lastChunk", SegmentMetadata::getLastChunk)
+                    .put("lastModified", SegmentMetadata::getLastModified)
+                    .put("firstChunkStartOffset", SegmentMetadata::getFirstChunkStartOffset)
+                    .put("lastChunkStartOffset", SegmentMetadata::getLastChunkStartOffset)
+                    .put("ownerEpoch", SegmentMetadata::getOwnerEpoch)
+                    .build();
+
+    public static final Map<String, Function<ReadIndexBlockMetadata, Object>> READ_INDEX_BLOCK_METADATA_FIELD_MAP =
+            ImmutableMap.<String, Function<ReadIndexBlockMetadata, Object>>builder()
+                    .put("name", ReadIndexBlockMetadata::getKey)
+                    .put("chunkName", ReadIndexBlockMetadata::getChunkName)
+                    .put("startOffset", ReadIndexBlockMetadata::getStartOffset)
+                    .put("status", ReadIndexBlockMetadata::getStatus)
+                    .build();
+
+    /**
+     * Append the given field name-value in a user-friendly format to the StringBuilder.
+     *
+     * @param builder The StringBuilder to append to.
+     * @param name    The name of the field.
+     * @param value   The value of the field.
+     */
+    public static void appendField(StringBuilder builder, String name, String value) {
+        builder.append(name).append("=").append(value).append(";");
+    }
+
+    /**
+     * Parse the given string into a map of keys and values.
+     *
+     * @param stringData The string to parse
+     * @return A map containing all the key-value pairs parsed from the string.
+     */
+    public static Map<String, String> parseStringData(String stringData) {
+        Map<String, String> parsedData = new HashMap<>();
+        List<String> fields = Arrays.asList(stringData.split(";"));
+        fields.forEach(kv -> {
+            List<String> pair = Arrays.asList(kv.split("="));
+            assert pair.size() == 2;
+            parsedData.put(pair.get(0), pair.get(1));
+        });
+        return parsedData;
+    }
+
+    /**
+     * Checks if the given key exists in the map. If it exists, it returns the value corresponding to the key and removes
+     * the key from the map.
+     *
+     * @param data The map to check in.
+     * @param key  The key.
+     * @return The value of the key if it exists.
+     * @throws IllegalArgumentException if the key does not exist.
+     */
+    public static String getAndRemoveIfExists(Map<String, String> data, String key) {
+        if (!data.containsKey(key)) {
```

Comment:
combine both of these into `.remove`. It will return `null` if it doesn't exist.

Reply:
I use `.remove` and then check for `null`.